4.0-Upgrade.md

```diff
@@ -0,0 +1,32 @@
+# Upgrading to chartmogul-ruby 4.0.0
+
+This new version replaces the existing pagination for the `.all()` endpoints that used a combination
+of `page` and `per_page` parameters, and instead uses a `cursor` based pagination. So to list
+(as an example) Plans you now can:
+
+```ruby
+ChartMogul.api_key = '<API key goes here>'
+
+# Getting the first page
+plans = ChartMogul::Plan.all(per_page: 12)
+
+# This will return an array of plans (if available), and a cursor + has_more fields
+{
+    "plans": [
+        {
+            "uuid": "some_uuid",
+            "data_source_uuid": "some_uuid",
+            "name": "Master Plan"
+        }
+    ],
+    "has_more": True,
```

Comment:
If this is an example JSON file of what the API returns WDYT about moving it to a different code block with `json` formatting?

Also, nitpick:
```suggestion
    "has_more": true,
```

Reply:
Nicely spotted, I copied this from Python, didn't notice it. We can move it, I was just copying it from what the other libraries were doing, but it's a good idea.